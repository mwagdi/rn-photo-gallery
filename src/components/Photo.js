import React from 'react';
import { TouchableOpacity, Text, Image, StyleSheet } from 'react-native';

export default ({ photo, openModal }) => (
    <TouchableOpacity onPress={() => openModal(photo)}>
        <Text style={styles.title}>{photo.title}</Text>
        <Image style={styles.image} source={{uri: photo.src}} />
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    title: {
        fontWeight: "bold",
        textAlign: "center"
    },
    image: {
        height: 100,
        width: 200
    }
})