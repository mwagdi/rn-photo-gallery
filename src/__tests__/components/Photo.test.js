import React from 'react';
import renderer from 'react-test-renderer';
import Photo from '../../components/Photo';

describe('<Photo />', () => {
    const photo = {
        id: 1,
        src: "http://example.com/example.jpg",
        title: "Image Title"
    }
    const tree = renderer.create(<Photo photo={photo} />).toJSON();
    it('has 2 children', () => {
        expect(tree.children.length).toBe(2);
    });
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});