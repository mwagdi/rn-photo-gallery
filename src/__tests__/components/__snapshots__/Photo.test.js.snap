// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photo /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Image Title
  </Text>
  <Image
    source={
      Object {
        "uri": "http://example.com/example.jpg",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 200,
      }
    }
  />
</View>
`;
